File: DentiSmart.API/DentiSmart.Domain/Models/Tratamiento.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace DentiSmart.Domain.Models
{
    [BsonIgnoreExtraElements]
    public class Tratamiento
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public DateTime FechaCreacion { get; set; } = DateTime.Now;
        [Required]
        public Cita Cita { get; set; }
        public Odontograma Odontograma { get; set; }
        public List<DetalleTratamiento> Detalle {get; set;}
    }
}

File: DentiSmart.App/app/src/main/java/com/dreamteam/dentismart/models/Tratamiento.kt
package com.dreamteam.dentismart.models

import java.io.Serializable

data class Tratamiento(
    val cita: Cita,
    val detalle: List<Detalle>,
    val fechaCreacion: String,
    val id: String,
    val odontograma: Odontograma
) : Serializable